### ISSUE TITLE
[java] UselessOverridingMethod: False negative when super call is wrapped in nested blocks

### ISSUE THREAD

**Issue by leemeii** [pmd/pmd#6491]
**Affects PMD Version:** 7.20.0

**Rule:**[UselessOverridingMethod](https://docs.pmd-code.org/latest/pmd_rules_java_design.html#uselessoverridingmethod)

**Description:**
The rule **UselessOverridingMethod** fails to detect redundant overrides when the super call is wrapped inside an additional nested block (braces {}).

**Code Sample demonstrating the issue:**

```java
public class UselessOverrideBug {

    // Scenario A: Standard useless override (PMD reports this correctly)
    public static class ClassA {
        @Override
        public String toString() {
            return super.toString();
        }
    }

    // Scenario B: Useless override wrapped in nested blocks (FN - PMD misses this)
    public static class ClassB {
        @Override
        public String toString() {
            { // Extra block level 1
              return super.toString();   
            }
        }
    }
}
```

**Expected outcome:**
PMD should report a violation for both ClassA and ClassB at their respective toString method declarations, as both methods do nothing but call the superclass implementation.This is a false-negative.

**Running PMD through:** *CLI*

**Comment by UncleOwen** [comment]
Another usecase for https://github.com/pmd/pmd/issues/6622

**Comment by UncleOwen** [comment]
Closing this issue in favor of https://github.com/pmd/pmd/pull/6639 which introduces a new rule to detect unnecessary blocks.
